deploy-tool/cert/k8_storage.go

```diff
@@ -0,0 +1,151 @@
+package cert
+
+import (
+	"context"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"regexp"
+	"strings"
+)
+
+type SecretStorage struct {
+	Namespace  string
+	KubeClient *kubernetes.Clientset
+}
+
+var matchLabels = map[string]string{
+	"manager": "magiccert",
+}
+var labelSelector = "manager=magiccert"
+
+var specialChars = regexp.MustCompile("[^a-zA-Z0-9_.-]+")
+
+func cleanKey(key string) string {
+	return fmt.Sprintf("cm.k8s.%v", specialChars.ReplaceAllString(key, ""))
+}
+
+var dataKey = "value"
+
+func (s *SecretStorage) Store(key string, value []byte) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   cleanKey(key),
+			Labels: matchLabels,
+		},
+		Data: map[string][]byte{
+			dataKey: value,
+		},
+	}
+
+	var err error
+
+	secretsApi := s.getSecretsAPI()
+	if s.Exists(key) {
+		_, err = secretsApi.Update(context.Background(), secret, metav1.UpdateOptions{})
+	} else {
+		_, err = secretsApi.Create(context.Background(), secret, metav1.CreateOptions{})
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SecretStorage) Load(key string) ([]byte, error) {
+	secretsApi := s.getSecretsAPI()
+
+	secret, err := secretsApi.Get(context.Background(), cleanKey(key), metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return secret.Data[dataKey], nil
```

Comment:
This should check if `secret.Data` contains `dataKey` and return an error if not.